src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -234,12 +243,16 @@ internal MetricPointStatus MetricPointStatus
             if (this.aggType != AggregationType.HistogramWithBuckets &&
                 this.aggType != AggregationType.Histogram &&
                 this.aggType != AggregationType.HistogramWithMinMaxBuckets &&
-                this.aggType != AggregationType.HistogramWithMinMax)
+                this.aggType != AggregationType.HistogramWithMinMax &&
+                this.aggType != AggregationType.Base2ExponentialHistogram &&
+                this.aggType != AggregationType.Base2ExponentialHistogramWithMinMax)
             {
                 this.ThrowNotSupportedMetricTypeException(nameof(this.GetHistogramSum));
             }
 
-            return this.mpComponents.HistogramBuckets.SnapshotSum;
+            return this.mpComponents.HistogramBuckets != null
```

Comment:
nit: This method can return based on this.aggtype.

if aggtype = his, return sum from it
if aggtype == exp.his, return sum from it.
esle throw.

Reply:
Yes agreed this would be the more natural way to have written this, and it is what I originally attempted 😆. However, the throw happens inside of method call `this.ThrowNotSupportedMetricTypeException(nameof(this.GetHistogramSum));`, so this wouldn't compile as is. That is you'd get a "not all paths return" error.

Anyways, I don't know how much value we're getting from pushing the exception handling into a separate method. Would probably read better without it.